srcCpp/perftest_publisher.cxx

```diff
@@ -1143,10 +1144,22 @@ int perftest_cpp::Subscriber()
     reader->WaitForWriters(_NumPublishers);
     announcement_writer->WaitForReaders(_NumPublishers);
 
+    /*
+     * Announcement message that will be used by the announcement_writer
+     * to send information to the Publisher. This message size will indicate
+     * different things.
+     *
+     * We will use 3 sizes: INITIALIZE_SIZE, LENGTH_CHANGED_SIZE and FINISHED_SIZE,
+     * msg.data will be used as the payload of such messages, so we choose the
+     * greatest size:
+     */
+    TestMessage announcement_msg;
+    announcement_msg.entity_id = _SubID;
+    announcement_msg.size = perftest_cpp::INITIALIZE_SIZE;
+    announcement_msg.data = new char[LENGTH_CHANGED_SIZE];
```

Comment:
LENGTH_CHANGED_SIZE is the greatest size of all 3. Alternatively we could do max(a,max(b,c))